### PR TITLE
Fixing invalid pull_request.sha

### LINKED ISSUE #200: Cannot read property 'head' of undefined
**Describe the bug**

During a run of the Github Actions I get the following:

```
Run stoplightio/spectral-action@v0.6.1
  with:
    file_glob: reference/v2/Index.yaml
    repo_token: ***
    event_name: pull_request
/usr/bin/docker run --name c27d31e4036e1dba5842deb4ae1a78b0ffbbb1_d82818 --label c27d31 --workdir /github/workspace --rm -e INPUT_FILE_GLOB -e INPUT_SPECTRAL_RULESET -e INPUT_REPO_TOKEN -e INPUT_EVENT_NAME -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/captains-log/captains-log":"/github/workspace" c27d31:e4036e1dba5842deb4ae1a78b0ffbbb1
(node:1) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'head' of undefined
    at extractSha (/action/dist/octokit.js:37:47)
    at /action/dist/octokit.js:53:34
    at /action/node_modules/fp-ts-contrib/lib/Do.js:31:96
    at Object.chain (/action/node_modules/fp-ts/lib/Either.js:474:57)
    at DoClass.bindL (/action/node_modules/fp-ts-contrib/lib/Do.js:31:43)
    at Object.exports.getRepositoryInfoFromEvent (/action/dist/octokit.js:53:6)
    at /action/dist/index.js:83:243
    at /action/node_modules/fp-ts/lib/EitherT.js:21:159
    at /action/node_modules/fp-ts/lib/Task.js:114:91
(node:1) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:1) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

**To Reproduce**

main.yml
```
name: Run Spectral Linter on Pull Requests

on:
  - pull_request

jobs:
  build:
    name: Run Spectral Linter
    runs-on: ubuntu-latest
    steps:
      # Check out the repository
      - uses: actions/checkout@v2

      # Run Spectral
      - uses: stoplightio/spectral-action@v0.6.1
        with:
          file_glob: 'reference/v2/Index.yaml'
```

**Additional context**
This also results in a 'pass' for the Github 'Check' which is surprising, I would expect an error on the command output to exit with 0 and show this as a failure.
image

### PR DESCRIPTION
**Pull request by grische**
It seems to be a regression of 8bb03cc4780934c62ee8ac062198493332150efe

Fixes #200

### CODE REVIEW

**Comment by hamermike** [comment]
@XVincentX Why was this not merged into a release? #200 is still an issue for myself.

**Comment by XVincentX** [comment]
@hamermike To be honest I've lost the memory of me closing this. I will try to take a look into and see if there's something we can do to fix it.

**Comment by hamermike** [comment]
I appreciate the fast response and thank you for looking into this!

**Comment by grische** [comment]
I am not convinced that this PR is the best approach here. While I think that `event: Event` is very important, the solution might need to look a bit different. Would be great if @XVincentX could have a look into that.

**Comment by XVincentX** [comment]
Generally speaking, this package needs some love — we've been neglecting it for a while.

Good to see some attention here. I'll try to check this out tomorrow and see what we can do to bring it back in shape.

File: src/octokit.ts [hunk 1]
```diff
@@ -41,10 +41,10 @@ export interface IRepositoryInfo {
   sha: string;
 }
 
-const extractSha = (eventName: string, event: any): E.Either<Error, string> => {
+const extractSha = (eventName: string, event: Event): E.Either<Error, string> => {
   switch (eventName) {
     case 'pull_request':
-      return E.right(event.pull_request.head.sha);
+      return E.right(event.after);
```
Comment (XVincentX) [inline src/octokit.ts:47]: The `after` property is effectively not available on pull request events. We'll have to find another way.

**Pull request closed without merging**